src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/ColumnDefinition.cs

```diff
@@ -511,6 +519,31 @@ internal void InternalTrimToSize()
 
         #region Internal Properties
 
+        /// <summary>
+        ///    Owner property.
+        /// </summary>
+        internal Grid Owner
+        {
+            get { return (_owner); }
+            set 
+            {
+                if(_owner != null && _owner != value)
+                {
+                    throw new ArgumentException(SR.Format(SR.GridCollection_InOtherCollection, "value", "ColumnDefinitionCollection"));
+                }
+                if(value == null)
+                {
+                    return;
```

Comment:
So if `Owner` is set to `null`, this will just ignore it and keep the `_owner` what it is. Given the first condition, `_owner` can either be `null` in which case this is fine, or it will be equal to `value`, in which case you add the logical children below second time. I would do

```C#
if (_owner != value)
{
    if (_owner != null) { remove children }
    _owner = value;
    if (value != null) { add children }
}

Reply:
Basically like I believe I said before, you need to consider the case where `value` is already equal the current value. `grid1.ColumnDefinitions = grid1.ColumnDefinitions` (i.e. on the same instance) should not throw.

Reply:
I had updated the code so that it does not throw an error if we do:
```grid1.ColumnDefinitions = grid1.ColumnDefinitions```